raft/raft.go

```diff
@@ -1392,6 +1408,9 @@ func (r *raft) removeNode(id uint64) {
 		return
 	}
 
+	// TODO(tbg): won't bad (or at least unfortunate) things happen if the
```

Comment:
Yeah, it's not ideal. But even if the leader steps down, the nodes that remain need to elect a new leader (which could have interactions with read leases, etc). So best practice would continue to be to transfer leadership away before removing the leader. But as a fallback, having the leader step down would probably be a good idea (unless we're missing something and this is already happening).